commands/utils/quick-pick-azure.ts

```diff
@@ -149,7 +149,9 @@ export async function confirmUserIntent(yesOrNoPrompt: string): Promise<boolean>
     let answer = await ext.ui.showInputBox(opt);
     answer = answer.toLowerCase();
     if (answer === 'yes') {
-        return answer === 'yes';
+        return true;
+    } else if (!cancelWhenNo && answer === 'no') {
+        return false;
```

Comment:
So what if they enter "true" or something?  Right now, the code would throw UserCancelled regardless of cancelWhenNo.

Suggest:

    } else if (!cancelWhenNo) {
  return false;
} else { 
        throw new UserCancelledError(); 
}

Reply:
The callers both look like this:
>     const should = await confirmUserIntent(`Are you sure?`);
>     if (should) {
>          … do some stuff
>     }

If user:
  types "yes" - does the work
  presses ESC - telemetry says it was cancelled
  types "no" - doesn't do the work, but telemetry says it did and that it succeeded

Seems to me the last two cases should be the same (in which case there's no need to return a Boolean. Otherwise, not sure what you would want to do differently.

Reply:
But then how you would recommend handling the options for a "pull from Azure". Right now handle a repo or an image. Both are the same command. 
Should I split the command into 2 instead?

Reply:
I don't understand the scenario.  Why do you want different behavior for ESC versus typing in No?

Reply:
Typing in Yes should only be for confirming a major operation such as a delete.  Simple yes/no should be yes/no buttons.

Reply:
I changed it, I agree with you. It is better to use this for  major operations